examples/hooks.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+# coding=utf-8
+"""
+A sample application for cmd2 demonstrating how to use hooks.
+
+This application shows how to use postparsing hooks to allow case insensitive
+command names, abbreviated commands, as well as allowing numeric arguments to
+follow a command without any intervening whitespace.
+
+"""
+
+import re
+
+from typing import List
+
+import cmd2
+
+
+class CmdLineApp(cmd2.Cmd):
+    """Example cmd2 application demonstrating the use of hooks.
+
+    This simple application has one command, `list` which generates a list
+    of 10 numbers. This command takes one optional argument, which is the
+    number to start on.
+
+    We have three postparsing hooks, which allow the user to enter:
+
+        (Cmd) list 5
+        (Cmd) L 5
+        (Cmd) l 5
+        (Cmd) L5
+        (Cmd) LI5
+
+    and have them all treated as valid input which prints a list of 10 numbers
+    starting with the number 5.
+    """
+
+    # Setting this true makes it run a shell command if a cmd2/cmd command doesn't exist
+    # default_to_shell = True
+    def __init__(self, *args, **kwargs):
+        # sneakily remove the cmd2.Cmd command called load
+        # this lets a user enter a command like "l5" and allows it to
+        # be unambiguous
+        delattr(cmd2.Cmd, "do_load")
+
+        super().__init__(*args, **kwargs)
+
+        # register three hooks
+        self.register_postparsing_hook(self.add_whitespace_hook)
```

Comment:
Interesting use case for multiple postparsing hooks.  I like the example, thanks for creating it.